#!/usr/bin/env python
PACKAGE = "aerial_robot_control"

from dynamic_reconfigure.parameter_generator_catkin import *
from aerial_robot_msgs.msg import DynamicReconfigureLevels

gen = ParameterGenerator()

gen.add("nmpc_flag", bool_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_FLAG, "NMPC Control Weight Flag", False)
gen.add("Qp_xy", int_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_Q_P_XY, "NMPC Qp_xy", 300, 0, 2000)
gen.add("Qp_z", int_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_Q_P_Z, "NMPC Qp_z", 400, 0, 2000)
gen.add("Qv_xy", int_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_Q_V_XY, "NMPC Qv_xy", 10, 0, 200)
gen.add("Qv_z", int_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_Q_V_Z, "NMPC Qv_z", 10, 0, 200)
gen.add("Qq_xy", int_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_Q_Q_XY, "NMPC Qq_xy", 300, 0, 2000)
gen.add("Qq_z", int_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_Q_Q_Z, "NMPC Qq_z", 500, 0, 2000)
gen.add("Qw_xy", int_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_Q_W_XY, "NMPC Qw_xy", 5, 0, 100)
gen.add("Qw_z", int_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_Q_W_Z, "NMPC Qw_z", 5, 0, 100)
gen.add("Qa", int_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_Q_A, "NMPC Qa", 1, 0, 10)
gen.add("Rt", int_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_R_T, "NMPC Rt", 1, 0, 10)
gen.add("Rac_d", int_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_R_AC_D, "NMPC Rac_d", 250, 0, 2000)

gen.add("i_gain_flag", bool_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_I_GAIN_FLAG, "NMPC I Gain Flag", False)
gen.add("i_gain_x", int_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_I_GAIN_X, "NMPC I Gain x", 3, 0, 20)
gen.add("i_gain_y", int_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_I_GAIN_Y, "NMPC I Gain y", 3, 0, 20)
gen.add("i_gain_z", int_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_I_GAIN_Z, "NMPC I Gain z", 3, 0, 20)
gen.add("i_gain_roll", double_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_I_GAIN_ROLL, "NMPC I Gain roll", 0.5, 0, 3)
gen.add("i_gain_pitch", double_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_I_GAIN_PITCH, "NMPC I Gain pitch", 0.5, 0, 3)
gen.add("i_gain_yaw", double_t, DynamicReconfigureLevels.RECONFIGURE_NMPC_I_GAIN_YAW, "NMPC I Gain yaw", 0.5, 0, 3)

exit(gen.generate(PACKAGE, "aerial_robot_control", "NMPC"))
